src/compiler/types.ts

```diff
@@ -8462,6 +8463,14 @@ namespace ts {
         readonly providePrefixAndSuffixTextForRename?: boolean;
         readonly includePackageJsonAutoImports?: "auto" | "on" | "off";
         readonly provideRefactorNotApplicableReason?: boolean;
+        readonly includeInlayParameterNameHints?: boolean;
+        readonly includeInlayNonLiteralParameterNameHints?: boolean;
+        readonly includeInlayDuplicatedParameterNameHints?: boolean;

```

Comment:
What are “duplicated” parameter name hints? Do we need all three of these options? What happens if I have `includeInlayParameterNameHints` disabled but the other two enabled? (Sorry if these questions have already been addressed.)

Reply:
Thanks for the review.

> What are “duplicated” parameter name hints?

The argument is an identifier and it's same as the parameter name. 
I'm not sure how to accurately describe this behavior. Any suggestions about the naming? 

```
function foo (a: number) {}
var a = 1
foo(a) 
```
> Do we need all three of these options?

[here's](https://github.com/microsoft/vscode/pull/113285#issuecomment-750868023) some context. I'm ok to remove them.

>disabled but the other two enabled?

There's no hints will be provided.

Reply:
I actually like the format that Cyrus showed where it was “**suppress** hints when ...” because it makes it more clear that those options won’t do anything if `includeInlayParameterNameHints` is disabled. Also, does VS Code let us show a nested/hierarchical checkbox list like that screenshot? I don’t think so, but that would be really nice.

Reply:
> suppress hints when ...

I guess that's the description what UI shows. Not about the implementations.
Another problem is VSCode would not open new feature's options by default.
If the option is `suppress something`, the default behavior is `not suppress something`, that is `include/provide something`. (If I'm correct).

> I don’t think so, but that would be really nice.

Sadly it's not supported.

Reply:
What about:

- includeInlay [xxxxx] HintsWhen [yyyyy]
- include [xxxxx]InlayHintsWhen [yyyyy]

Reply:
I think my ideal scenario would be

```ts
includeInlayParameterNameHints?: "none" | "literals" | "all";
suppressInlayParameterNameHintsWhenArgumentMatchesName?: boolean;
```

and the latter can default to `true` both in VS Code and in TS Server. But I’m ok with swapping it to `includeInlayParameterNameHintsWhenArgumentMatchesName` in the options, and maybe trying to use “suppress” language in the VS Code UI and invert the value sent. I just really don’t want the options UI to allow for a bunch of checkboxes that sound like they should enable something but actually do nothing because they’re really sub-options of `includeInlayParameterNameHints`.